### PR TITLE
Fix matching-service frontend bugs

### PR DESCRIPTION
**Pull request by ongweisheng**
- Fix able to not select difficulty and find match (Must select difficulty now before finding match)
- Fix unable to end interview if the other user ended the interview first (Able to end interview and go back to home page even if the other user ends the interview first)

### CODE REVIEW

File: frontend/src/components/HomePage.js [hunk 1]
```diff
@@ -49,7 +49,11 @@ const HomePage = () => {
 
     const findMatch = (event) => {
         event.preventDefault();
-        navigate("/search", { state: { difficulty: difficulty, username: user } });
+        if (difficulty === "") {
+            setAlertMessage("Difficulty level not selected. Select difficulty before finding match!")
+        } else {
+            navigate("/search", { state: { difficulty: difficulty, username: user } })
+        }
```
Comment (TheSpaceCuber) [inline frontend/src/components/HomePage.js:56]: Alternatively default difficulty can be easy, but this works too.

File: frontend/src/components/InterviewPage.js [hunk 2]
```diff
@@ -31,12 +31,29 @@ const InterviewPage = () => {
...
+                    console.log("Interview terminated from other user")
+                    setQuestionTitle("")
+                    setQuestionDescription("")
+                    setShowEndInterview(false)
+                    navigate("/home")
+                } else {
+                    MatchingService
+                        .deleteInterview(interviewId)
+                        .then((res) => {
+                            setQuestionTitle("")
```
Comment (hiongkaihan) [inline frontend/src/components/InterviewPage.js:48]: LGTM
However not sure if we should standardize using async/await vs using promises

**Review by Herrekt — CHANGES_REQUESTED** [review]
Tested matching however after the timer runs out. I was not able to cancel matchmaking when "Unable to find match" comes out and return to the question difficulty selection menu. It just stays stuck on the menu although I clicked and confirm cancel find match.
In console, it returns: 
"
CS2106 waiting for match, timer at 31 seconds
No interview found for: CS2106
Error 500: Unable to delete match request in DB as no match entry found for user: CS2106
"

**Review by Herrekt — APPROVED** [review]
LGTM!

**Pull request merged**